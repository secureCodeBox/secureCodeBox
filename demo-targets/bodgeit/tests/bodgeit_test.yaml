suite: Full Snapshot
templates:
  - deployment.yaml
  - service.yaml
  - ingress.yaml
  - NOTES.txt
  - ../templates/tests/test-connection.yaml

tests:
  - it: matches the snapshot
    set:
      labels: {foo: bar}
      annotations: {foo: bar}
      ingress.enabled: true
    asserts:
      - matchSnapshot: {}


