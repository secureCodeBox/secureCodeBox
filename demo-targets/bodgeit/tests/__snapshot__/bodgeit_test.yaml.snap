# SPDX-FileCopyrightText: the secureCodeBox authors
#
# SPDX-License-Identifier: Apache-2.0

matches the snapshot:
  1: |
    raw: |2
      1. Get the application URL by running these commands:
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        foo: bar
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: bodgeit-0.0.0
      name: bar
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: foo
      template:
        metadata:
          annotations:
            foo: bar
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: foo
        spec:
          affinity:
            nodeAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                nodeSelectorTerms:
                  - matchExpressions:
                      - key: foo
                        operator: In
                        values:
                          - bar
                          - baz
          containers:
            - image: docker.io/securecodebox/bodgeit:latest
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: http
                initialDelaySeconds: 60
              name: bodgeit
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /
                  port: http
                initialDelaySeconds: 60
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 60
              resources:
                requests:
                  cpu: 123m
                  memory: 123Mi
              securityContext:
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1234
          imagePullSecrets:
            - name: foo
          nodeSelector:
            foo: bar
          securityContext:
            fsGroup: 1234
          tolerations:
            - effect: NoSchedule
              key: foo
              operator: Exists
              value: value1
  3: |
    kind: Ingress
    metadata:
      annotations:
        foo: bar
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: bodgeit-0.0.0
      name: bar
    spec:
      rules:
        - host: chart-example.local
          http:
            paths: null
      tls:
        - hosts:
            - chart-example.local
          secretName: chart-example-tls
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        foo: bar
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: bodgeit-0.0.0
      name: bar
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: foo
      type: ClusterIP
  5: |
    apiVersion: v1
    kind: Pod
    metadata:
      annotations:
        helm.sh/hook: test-success
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: foo
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: bodgeit-0.0.0
      name: bar-test-connection
    spec:
      containers:
        - args:
            - bar:8080
          command:
            - wget
          image: busybox
          name: wget
      restartPolicy: Never
